pkg/kp/consulutil/watch.go

```diff
@@ -22,6 +22,7 @@ func WatchPrefix(
 	done <-chan struct{},
 	outErrors chan<- error,
 ) {
+	// TODO return this for closing on the client side
 	defer close(outPairs)
```

Comment:
my understanding is that it's not idiomatic to close a channel you were passed as an argument, it would make more sense for this function to do its work in a goroutine and return a channel it created